components/query_filter/utils_unittest.cc

```diff
@@ -0,0 +1,101 @@
+                GURL("https://test.com/?gclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  // Expect filtering `gclid` param when cross origin `redirect_source_url`
+  // and same origin `initiator_url`
+  EXPECT_EQ(query_filter::MaybeApplyQueryStringFilter(
+                GURL("https://test.com"), GURL("https://brave.com"),
+                GURL("https://test.com/?gclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  // Expect filtering `fbclid` param when cross origin `initiator_url`
+  // and invalid `redirect_source_url`
+  EXPECT_EQ(query_filter::MaybeApplyQueryStringFilter(
+                GURL("https://brave.com"), GURL(),
+                GURL("https://test.com/?fbclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  // Expect filtering `fbclid` param when cross origin `initiator_url`
+  // and invalid `redirect_source_url`
+  // (even though `internal_redirect=true`)
+  EXPECT_EQ(query_filter::MaybeApplyQueryStringFilter(
+                GURL("https://brave.com"), GURL(),
+                GURL("https://test.com/?fbclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  // Expect filtering `fbclid` param when cross origin `initiator_url`
+  // and invalid `redirect_source_url` even though `internal_redirect` == true
+  EXPECT_EQ(query_filter::MaybeApplyQueryStringFilter(
+                GURL("https://brave.com"), GURL(),
+                GURL("https://test.com/?fbclid=123"), "GET", false),
+            GURL("https://test.com/"));
+  // Expect filtering `mkt_tok` param when cross origin `redirect_source_url`
+  // and invalid `initiator_url`
+  EXPECT_EQ(query_filter::MaybeApplyQueryStringFilter(
+                GURL(), GURL("https://brave.com"),
+                GURL("https://test.com/?mkt_tok=123"), "GET", false),
+            GURL("https://test.com/"));
+  // Test filtering `gclid` param and leaving other params
+  EXPECT_EQ(
+      query_filter::MaybeApplyQueryStringFilter(
+          GURL("https://brave.com"), GURL(),
+          GURL("https://test.com/?gclid=123&unsubscribe=123"), "GET", false),
+      GURL("https://test.com/?unsubscribe=123"));
+  // Test filtering `gclid` param and leaving other params
+  EXPECT_EQ(
+      query_filter::MaybeApplyQueryStringFilter(
+          GURL(), GURL("https://brave.com"),
+          GURL("https://test.com/?gclid=123&Unsubscribe=123"), "GET", false),
+      GURL("https://test.com/?Unsubscribe=123"));
+  // Test nil when nothing is needed to be filtered
+  EXPECT_FALSE(query_filter::MaybeApplyQueryStringFilter(
+      GURL("https://brave.com"), GURL("https://brave.com"),
+      GURL("https://test.com/"), "GET", false));
+  // Test nothing is filtered with invalid request url
+  EXPECT_FALSE(query_filter::MaybeApplyQueryStringFilter(
+      GURL("https://brave.com"), GURL("https://brave.com"), GURL(), "GET",
+      false));
+  // Only `GET` requests are filtered
```

Comment:
For these tests to be more useful, we need the `request_url` to include a query param that would get stripped. So I'd replace `https://test.com/` with `https://test.com/?gclid=123` in this section.

As they are currently written, the `request_url` would not be changed even if it qualified for URL modifications.

Reply:
Good call

Reply:
Done